modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,728 @@
+	$post_attachments   = array ();
+	while ( $sub_posts = array_splice( $attachment_parents, 0, 100 ) ) {
+		$post_parents = implode( ',', array_map( 'intval', $sub_posts ) );
+
+		// Get the attachment IDs for all posts. We need to see how many
+		// attachments each post parent has and limit it to 5.
+		$query                = "SELECT ID, post_parent FROM {$wpdb->posts} WHERE post_parent IN ({$post_parents}) AND post_type='attachment' AND post_mime_type='image/jpeg' LIMIT 0,1000;";
+		$all_attachments      = $wpdb->get_results( $query );
+		$selected_attachments = array ();
+		$attachment_count     = array ();
+
+		foreach ( $all_attachments as $attachment ) {
+			if ( ! isset( $attachment_count[ $attachment->post_parent ] ) ) {
+				$attachment_count[ $attachment->post_parent ] = 0;
+			}
+
+			// Skip this particular attachment if we already have 5 for the post
+			if ( $attachment_count[ $attachment->post_parent ] >= 5 ) {
+				continue;
+			}
+
+			$selected_attachments[] = $attachment->ID;
+			$attachment_count[ $attachment->post_parent ] ++;
+		}
+
+		// bail if there weren't any attachments to avoid an extra query
+		if ( empty( $selected_attachments ) ) {
+			continue;
+		}
+
+		// Get more of the attachment object for the attachments we actually care about
+		$attachment_ids   = implode( ',', array_map( 'intval', $selected_attachments ) );
+		$query            = "SELECT p.ID, p.post_parent, p.post_title, p.post_excerpt, p.guid FROM {$wpdb->posts} as p WHERE p.ID IN ({$attachment_ids}) AND p.post_type='attachment' AND p.post_mime_type='image/jpeg' LIMIT 500;";
+		$attachments      = $wpdb->get_results( $query );
+		$post_attachments = array_merge( $post_attachments, $attachments );
+	}
+
+	unset( $initstr );
+	$latest_mod = '';
+	foreach ( $posts as $post ) {
+
+		/**
+		 * Filter condition to allow skipping specific posts in sitemap.
+		 *
+		 * @module xml-sitemap
+		 *
+		 * @since 3.9
+		 *
+		 * @param bool $skip Current boolean. False by default, so no post is skipped.
+		 * @param WP_POST $post Current post object.
+		 */
+		if ( apply_filters( 'jetpack_sitemap_skip_post', false, $post ) ) {
+			continue;
+		}
+
+		$post_latest_mod = null;
+		$url             = array ( 'loc' => esc_url( get_permalink( $post->ID ) ) );
+
+		// If this post is configured to be the site home, skip since it's added separately later
+		if ( untrailingslashit( get_permalink( $post->ID ) ) == untrailingslashit( get_option( 'home' ) ) ) {
+			continue;
+		}
+
+		// Mobile node specified in http://support.google.com/webmasters/bin/answer.py?hl=en&answer=34648
+		$url['mobile:mobile'] = '';
+
+		// Image node specified in http://support.google.com/webmasters/bin/answer.py?hl=en&answer=178636
+		// These attachments were produced with batch SQL earlier in the script
+		if ( ! post_password_required( $post->ID ) && $attachments = wp_filter_object_list( $post_attachments, array ( 'post_parent' => $post->ID ) ) ) {
+
+			$url['image:image'] = array ();
```

Comment:
Coding standards: I believe it should be `array()`. There are a handful of these.